firestore/report.go

```diff
@@ -0,0 +1,76 @@
+package firestore
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/firestore"
+
+	"github.com/benchttp/server/benchttp"
+)
+
+// Ensure service implements interface.
+var _ benchttp.ReportService = (*ReportService)(nil)
+
+type ReportService struct {
+	client       *firestore.Client
+	collectionID string
+}
+
+// NewReportService returns a new repository wrapping a
+// Firestore client. The internal client uses the given project.
+// The repository only uses collectionID as the collection to
+// create documents to and retrieve documents from.
+func NewReportService(ctx context.Context, projectID, collectionID string) (*ReportService, error) {
```

Comment:
[suggestion] I'd avoid pointers returning when possible
```suggestion
func NewReportService(ctx context.Context, projectID, collectionID string) (ReportService, error) {
```